### ISSUE TITLE
Subscribe to channel on a server

### ISSUE THREAD

**Issue by Danielv123** [ClusterWS/ClusterWS#88]
### Submitting

- [ ] Bug
- [x] Question
- [ ] Suggetion
- [ ] Other

I am using the latest version of clusterWS on Windows, but my target system is Ubuntu.

From the documentation on https://github.com/ClusterWS/ClusterWS/wiki/Pub-Sub-System I see that my server can push events to any channel. It can also talk 1on1 to any client that connects to that worker.

The application I am working on is a matchmaking browser. For this I need to subscribe to an incoming channel, not just a single socket (which could end up on the wrong worker), but this does not seem to be possible.

What is the correct way to go about this? Should I find a way to not listen to a channel, should I import the client library and connect to my own server or is there something I am missing?

**Comment by goriunov** [comment]
The channels are designed to have more then just 1 subscriber (you can have only one subscriber as well). It is does not matter on which worker will the channel end up as pub sub system is designed to be able to send message from any server to the right channel. 

One way to listen on the publish event on the server is `onPublish` event documentation is in [here](https://github.com/ClusterWS/ClusterWS/wiki/Middleware-Functions). However i am not sure it will work out well as publish system will not process with message if no client on that worker subscribed to that channel. (i need more information about the app and how you plan it to work)

Second way is to use `onMessageFromWorker` from the same documentation in [here](https://github.com/ClusterWS/ClusterWS/wiki/Middleware-Functions) and standard  1on1  communication with client.  For example:
``` js
function Worker() { 
    const wss = this.wss
    const server = this.server 
     
    // listen on messages from workers
     wss.setMiddleware('onMessageFromWorker', (message) => {
          // execute any code on message form workers
     })

    // Listen on connections to websocket server
    wss.on('connection', (socket) => {
        socket.on('matchmaking', (message) => {
          // send message to all workers  NOTE: include itself
           wss.publishToWorkers(message)
         })
    })
}
``` 

That is actually a good point you have provided, i will add functionality to subscribe workers to the channels (only with ability to listen on messages which are send through the channels)

**Comment by Danielv123** [comment]
Yeah, I found the publishToWorkers while toying around with the problem. It works, but it feels like a dirty hack and its not very readable. I am glad this functionality is coming though 👍

**Comment by goriunov** [comment]
I have released new version 3.1.0 the functionality to listen on channels is added there. Check [this](https://github.com/ClusterWS/ClusterWS/wiki/Pub-Sub-System) for more info about it.

**Comment by goriunov** [comment]
Close issue as completed.
